### DISCUSSION TITLE
Access current URL  ( window.location.href)

### DISCUSSION THREAD

**Discussion by artificial-cassiano** [coder/code-server#4644] [Q&A]
Hi, 
I want to get some custom parameters from the URL, can I access the window location by an Environment variable?

so, for code-server, if I have the following url:
```
http://localhost:8443?parameter0=lucas
```

how can I access `parameter0`?

**Comment by jsjoeio** [comment]
There is not an environment variable. But you could write a Web Extension and use that to grab the params from the URL I bet. Here are some [docs](https://code.visualstudio.com/api/extension-guides/web-extensions)

**Reply by jsjoeio** [reply]
Here is an example: https://github.com/microsoft/vscode-extension-samples/tree/main/helloworld-web-sample

**Reply by alanhe421** [reply]
So what method can be used in vscode to get the URL?

**Reply by code-asher** [reply]
I believe you can use `asExternalUri` https://github.com/coder/code-server/issues/1571#issuecomment-1828482849

**Reply by code-asher** [reply]
To clarify, `asExternalUri` will get you a tunnel to view a certain port.  It may or may not be the URI that is being used to access code-server; there is no guarantee it is the same.

**Reply by code-asher** [reply]
If you need the query params, we write those into `~/.local/share/code-server/coder.json` so you could read that file to get them.

**Reply by alanhe421** [reply]
My code-server is bind to 0.0.0.0:8080. Eventually, I configured a reverse proxy to expose the web service, for example, https://xxxx.x/xx?folder=helloworld.

I hope to be able to get the current URL, such as https://xxxx.x/xx?folder=helloworld.

I'm not sure if there is a way.

**Reply by code-asher** [reply]
I see, that information is not exposed to plugins as far as I know.  `asExternalUri` might work for your use case.

**Reply by alanhe421** [reply]
![image](https://github.com/coder/code-server/assets/9245110/6838904c-e7b6-4f6f-b6a6-8534ddae8c1c)


I think should be https://127.0.0.1:3001/xyz/stable-695af097c7bd098fbf017ce3ac85e09bbc5dda06/callback?vscode-reqid%3D1%26vscode-scheme%3Dcode-oss%26vscode-path%3D%252F

## code
```typescript
  vscode.env.asExternalUri(vscode.Uri.parse(vscode.env.uriScheme + ':/')).then(res => {
        vscode.window.showInformationMessage(res.toString());
    });
```

## caddy proxy

```caddyfile
127.0.0.1:3001 {
	route /xyz* {
		uri strip_prefix /xyz
		reverse_proxy 127.0.0.1:8080
	}
}

```

**Reply by code-asher** [reply]
Oh yeah you are right.  Looks like a bug.  We should be prefixing the base path.

**Reply by alanhe421** [reply]
So do I need to submit another issue? I hope to follow up on the resolution of this problem.

**Reply by code-asher** [reply]
Yeah I think a separate issue is a good idea, that will help make sure we do not lose track of this.

**Reply by code-asher** [reply]
Seems like we do prepend the base though so that is weird: 

https://github.com/coder/code-server/blob/d0215eca8804d0be5d3551c39404e336032c3141/patches/proxy-uri.diff#L78
